apply/apply.go

```diff
@@ -43,22 +53,56 @@ func NewApplierFromFile(clusterfile string) (Interface, error) {
 	return NewApplier(cluster)
 }
 
-func GetClusterFromFile(filepath string) (cluster *v1.Cluster, err error) {
-	cluster = &v1.Cluster{}
-	if err = utils.UnmarshalYamlFile(filepath, cluster); err != nil {
-		return nil, fmt.Errorf("failed to get cluster from %s, %v", filepath, err)
-	}
-	cluster.SetAnnotations(common.ClusterfileName, filepath)
-	return cluster, nil
-}
-
 func NewApplier(cluster *v1.Cluster) (Interface, error) {
 	switch cluster.Spec.Provider {
 	case common.AliCloud:
 		return NewAliCloudProvider(cluster)
 	case common.CONTAINER:
 		return NewAliCloudProvider(cluster)
 	}
-
 	return NewDefaultApplier(cluster)
 }
+
+func NewAliCloudProvider(cluster *v1.Cluster) (Interface, error) {
+	k8sClient, err := k8s.Newk8sClient()
+	if err != nil {
+		logger.Warn(err)
+	}
+
+	return &cloud.Applier{
+		ClusterDesired: cluster,
+		Client:         k8sClient,
+	}, nil
+}
+
+func NewDefaultApplier(cluster *v1.Cluster) (Interface, error) {
+	imgSvc, err := image.NewImageService()
+	if err != nil {
+		return nil, err
+	}
+
+	fs, err := filesystem.NewFilesystem()
+	if err != nil {
+		return nil, err
+	}
+
+	gs, err := guest.NewGuestManager()
+	if err != nil {
+		return nil, err
+	}
+
+	k8sClient, err := k8s.Newk8sClient()
+	if err != nil {
+		logger.Warn(err)
```

Comment:
Same issue. We should handle the detailed error as early as possible. Otherwise, we insert a meaningless or nil k8sClient into the local.Applier. We let the virus run in a later time. Definitely it brings burden for code. 

I am wondering if I have the correct understanding.